output/report.go

```diff
@@ -24,31 +24,37 @@ import (
 	"fmt"
 
 	"github.com/yarpc/crossdock/execute"
+	"github.com/yarpc/crossdock/plan"
 )
 
 // Reporter is responsible for outputting test results
 type Reporter interface {
-	Stream(tests <-chan execute.TestResponse) Summary
-}
-
-// ReporterFunc streams test results to the console
-type ReporterFunc func(tests <-chan execute.TestResponse) Summary
-
-// Stream allows ReporterFunc to fulfill the Reporter interface
-func (f ReporterFunc) Stream(tests <-chan execute.TestResponse) Summary {
-	return f(tests)
+	Start(config *plan.Config) error
+	Next(response execute.TestResponse, config *plan.Config)
+	End(config *plan.Config) (Summary, error)
 }
 
 // GetReporter returns a ReporterFunc for the given name
 func GetReporter(name string) (Reporter, error) {
 	// default reporter
 	if name == "" {
-		return List, nil
+		return &List{}, nil
 	}
 	// else a specific value was provided
 	switch name {
 	case "list":
-		return List, nil
+		return &List{}, nil
+	case "json":
+		return &JSON{}, nil
+	case "list,json":
+		fallthrough
+	case "json,list":
```

Comment:
Yeah, as an expedient. Maybe a temporary expedient, but need to give TTL attention before making this nice.